### PR TITLE
Fixed bug with LAST_NAME & Added more field data

### PR DESCRIPTION
**Pull request by Joe-Good**
Added some TODOs

### CODE REVIEW

File: pages/home/home_page.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def tap_submit(self):
 
     def enter_last_name(self, last_name):
         self.clear_text_from(HomePageLocators.LAST_NAME)
-        self.send_text_to(HomePageLocators.FIRST_NAME, last_name)
+        self.send_text_to(HomePageLocators.LAST_NAME, last_name) #Changed from FIRST_NAME - #bugfix
```
Comment (GreenJeans) [inline pages/home/home_page.py]: Sweet!  You can probably remove the comment :)

File: pages/home/home_page_locators.py [hunk 2]
```diff
@@ -8,4 +8,11 @@ class HomePageLocators(object):
...
     PHONE_NUMBER = (By.ID, "Phone")
     SUBMIT_BUTTON = (By.XPATH, "//input[@type = 'button'][@value = 'Continue']")
+    EMPLOYEES_SITE = (By.ID, "Employees")
+    COMPANY_ADDRESS = (By.ID, "Address1")
+    CITY_NAME = (By.ID, "City")
+    COUNTRY_NAME = (By.ID, "Country")
+    STATE_NAME = (By.ID, "State")
+    COMPANY_ZIP = (By.ID, "Zip")
     FINAL_SUBMIT = (By.ID, "btnSubmit")
+    QUESTIONS_COMMENTS = (By.ID, "Questions") #Take this opportunity to verify maximum char limit of 255 by RAND 256
```
Comment (GreenJeans) [inline pages/home/home_page_locators.py]: This comment is to keep a note to make some test cases later, yeah?  Maybe time to make a notebook of tests you intend to write? It'll be hard to find them in here later.

File: tests/test_unittest.py [hunk 3]
```diff
@@ -20,6 +20,9 @@ def test_incomplete_login(self):
         self.home_page.tap_submit()
 
         # Test Case: Assert error is present if user attempts to submit form prematurely
+        # TODO: Replace with (bool?) validator that will fail the case if unable to fill a field
+        # TODO: Cont'd Use that to drive the flow control: e.g.
+        # TODO: Try to fill COMPANY_ADDRESS field before it is available. If that fails, you win!
```
Comment (GreenJeans) [inline tests/test_unittest.py]: what? You wondering how to make the validator do something? I don't follow.

**Comment by GreenJeans** [comment]
Looks fine! just curious about some comments.

File: pages/home/home_page.py (hunk 1 above)
Comment (Joe-Good) [inline pages/home/home_page.py]: Done in my local -- will add a new tip.

File: pages/home/home_page_locators.py (hunk 2 above)
Comment (Joe-Good) [inline pages/home/home_page_locators.py]: Oh yeah, I was planning to do this today. Just creating visibility. For now, I just want a bit more flow control, and to do a couple more things with this particular page. I will at least pseudocode the cases I want into the test_unittest.py file today.

File: tests/test_unittest.py (hunk 3 above)
Comment (Joe-Good) [inline tests/test_unittest.py]: I understand your confusion. I think I can use the absence of a field being available as a form of validation -- in this case, the COMPANY_ADDRESS field won't be around until they do the first page of fields correctly. If I can have a short bit of logic where Succeeding at filling in that field Fails the case (if it shouldn't be there). Obviously there are many ways we can control the flow, I'm just curious if you have an idea of the best "Pythonic" way to do this. I'll get to work trying to phrase this in Pseudocode.

Comment (GreenJeans) [inline tests/test_unittest.py]: You don't need to check if you can fill the field, you can just look on the page to see if it's there.  You can use the BasePage _get_element method, it should return a NoSuchElement (or similar) exception if the element isn't found using the locator you supply

Comment (GreenJeans) [inline tests/test_unittest.py]: Oh, maybe better would be to add that same method to the BaseValidator with a better name...something like "def element_exists(locator):" and "def element_does_not_exist(locator):"

Comment (Joe-Good) [inline tests/test_unittest.py]: Typical Joe...create a dozen lines of code when 2 will do! I'll see if I can implement what you suggested.

File: tests/test_unittest.py [hunk 4]
```diff
@@ -12,14 +12,34 @@ def setup_class(self, home_page, home_page_validator, data):
...
         self.home_page.tap_submit()
+        prematurely_completed = False
+        try: #perhaps if?
+            self.home_page.enter_company_address(self.data["companyAddress"])
+            print("1. Company Address should NOT be on screen - FAIL")
+            prematurely_completed = True
+        except: #perhaps else?
+            if prematurely_completed:
+                print("2. This should have failed - FAIL")
+                #fail test case somehow
```
Comment (GreenJeans) [inline tests/test_unittest.py]: You can force assertions with 'Assert 0' or similar...not that I'm suggesting it here :)

**Pull request merged**